compiler/src/main/java/com/squareup/anvil/compiler/ModuleMerger.kt

```diff
@@ -92,112 +84,107 @@ internal class ModuleMerger(
         // modules contain the same bindings and are contributed to the same scope. To avoid this
         // issue we filter all generated Anvil modules except for the one that was generated for
         // this specific class.
-        val fqName = it.fqNameSafe
+        val fqName = it.fqName
         !fqName.isAnvilModule() || fqName == anvilModuleName
       }
-      .mapNotNull {
-        val contributesAnnotation =
-          it.annotationOrNull(contributesToFqName, scope = scopeFqName)
-            ?: return@mapNotNull null
-        it to contributesAnnotation
+      .mapNotNull { contributedClass ->
+        val contributesAnnotation = contributedClass.annotationOrNull(
+          annotationName = contributesToFqName,
+          scopeName = scopeFqName
+        ) ?: return@mapNotNull null
+
+        contributedClass to contributesAnnotation
```

Comment:
Creating this pair is unnecessary now. You can just return the annotation, because it keeps a reference to its class.